async/amqp/amqp.go

```diff
@@ -44,13 +44,13 @@ func (m *message) Decode(v interface{}) error {
 
 func (m *message) Ack() error {
 	err := m.del.Ack(false)
-	trace.SpanSuccess(m.span)
+	defer trace.SpanComplete(m.span, nil)
```

Comment:
Hi, the ticket was about only to introduce `SpanComplete()` and replace some of the cases where we can use it.
`SpanSuccess` and `SpanError` will stay to make the code more readable and not use nil.

since we are returning right after finishing a span, defer does not make much sense.